js/InputReader.js

```diff
@@ -0,0 +1,64 @@
+export class InputReader {
+    #input
+
+    constructor(input) {
+        if (typeof(input) !== "string") {
+            throw new TypeError("Expected input to be a string.")
+        }
+
+        this.#input = input
+    }
+
+    /**
+     * Iterates through the input string and yields all contained tokens.
+     *
+     * A token is either a Number representing an operand,
+     * or a string representing an operator or a bracket.
+     * E.g. the input "(3+45)" yields: "(", 3, "+", 45, ")"
+     */
+    *iterateTokens() {
+        for (let i = 0; i < this.#input.length; i++) {
+            if (InputReader.#isNumber(this.#input[i])) {
+                let number = InputReader.#completeNumberParser(this.#input, i)
+                i += InputReader.#lengthOfNumber(number) - 1
+                yield parseInt(number)
```

Comment:
Why parse the number again?

Reply:
Leftovers from when I wasn't sure #completeNumberParser returned a number.